app/lib/handlers/transaction_intent_request_handler.rb

```diff
@@ -51,13 +51,37 @@ def add_items
           title: 'Added Items',
           content: Helpers::HandlerHelper.prepare_items_for_card_without_date(categorized_items)
         },
+        reprompt: 'Is there anything else I can help you with?',
         end_session: true
       )
     end
 
-    # TODO, this will be implemented later
     def remove_items
-
+      removed_items = items.map do |k,v|
+        item = Item.owned_by(user_id).where(name: k).order(:created_at).first
+        item_meta = {name: k, quantity: v}
+        while item.present? && v > 0 do
+          v -= item.quantity
+          item.quantity = (item.quantity - v >= 0) ? item.quantity - v : 0
+          if item.quantity == 0
+            item.destroy
+            item = Item.owned_by(user_id).where(name: k).order(:created_at).first
+          end
+        end
+        item_meta[:quantity] -= v if v.positive?
+        item.save if item.present?
+        item_meta
+      end
```

Comment:
This seems a lot to have packed into a single method.  I'd refactor this if possible